tests/python/pants_test/repo_scripts/test_git_hooks.py

```diff
@@ -0,0 +1,161 @@
+ERROR: All '__init__.py' files should be empty or else only contain a namespace
+declaration, but the following contain code:
+---
+subdir/__init__.py
+""")
+
+      # Check that a valid __init__.py succeeds.
+      safe_file_dump(init_py_path, '')
+      self._assert_subprocess_success(worktree, [package_check_script, 'subdir'])
+      safe_file_dump(init_py_path,
+                     "__import__('pkg_resources').declare_namespace(__name__)")
+      self._assert_subprocess_success(worktree, [package_check_script, 'subdir'])
+
+  def test_added_files(self):
+    get_added_files_script = os.path.join(self.pants_repo_root,
+                                          'build-support/bin/get_added_files.sh')
+    with self._create_tiny_git_repo() as (git, worktree, _):
+      # Create a new file.
+      new_file = os.path.join(worktree, 'wow.txt')
+      safe_file_dump(new_file, '')
+      # Stage the file.
+      rel_new_file = os.path.relpath(new_file, worktree)
+      git.add(rel_new_file)
+      self._assert_subprocess_success_with_output(
+        worktree, [get_added_files_script],
+        # This should be the only entry in the index, and it is a newly added file.
+        full_expected_output="{}\n".format(rel_new_file))
+
+  def test_check_headers(self):
+    header_check_script = os.path.join(self.pants_repo_root,
+                                       'build-support/bin/check_header_helper.py')
+    with self._create_tiny_git_repo() as (_, worktree, _):
+      new_py_path = os.path.join(worktree, 'subdir/file.py')
+
+      # Check that a file with an empty header fails.
+      safe_file_dump(new_py_path, '')
+      self._assert_subprocess_error_with_input(
+        worktree, [header_check_script, 'subdir'],
+        # The python process reads from stdin, so we have to explicitly pass an empty string in
+        # order to close it.
+        input='',
+        expected_excerpt="""\
+subdir/file.py: failed to parse header at all
+""")
+
+      # Check that a file with a random header fails.
+      safe_file_dump(new_py_path, 'asdf')
+      self._assert_subprocess_error_with_input(
+        worktree, [header_check_script, 'subdir'],
+        input='',
+        expected_excerpt="""\
+subdir/file.py: failed to parse header at all
+""")
+      # Check that a file without a valid copyright year fails.
+      cur_year_num = datetime.datetime.now().year
+      cur_year = str(cur_year_num)
+      safe_file_dump(new_py_path, """\
+# coding=utf-8
+# Copyright YYYY Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+""")
+      self._assert_subprocess_error_with_input(
+        worktree, [header_check_script, 'subdir'],
+        input='',
+        expected_excerpt="""\
+subdir/file.py: copyright year must match '20\\d\\d' (was YYYY): current year is {}
```

Comment:
Aha, I wasn't properly groking what `input` did here. Maybe rename `input` to `added_files` here to make it more clear what this is doing on a domain-level?

Reply:
That makes sense! I will probably keep the `._assert_*` call the same but add a variable.

Reply:
I did not! I added a local method to make the domain-specific bit more clear as I have heard those are all the rage in this newfangled python thing.